### DISCUSSION TITLE
How to debug txs with msys2 and cmake?

### DISCUSSION THREAD

**Discussion by octaeder** [texstudio-org/texstudio#2919] [Q&A]
It seems that I'm no longer able to do a debug build (I use ``qDebug()<<"...";`` at the beginning of Texstudio::Texstudio() and see no output). The manual says ``qt-cmake .. -DCMAKE_BUILD_TYPE=Debug``. But msys2 doesn't find qt-cmake. Using ``cmake ../texstudio/ -DCMAKE_BUILD_TYPE=Debug`` doesn't work either. But the cache reflects this setting. Is there any chance that this will work?

**Comment by sunderme** [comment]
building on msys2 with debug works.
maybe you need to start with a clean build folder ?

**Comment by octaeder** [comment]
You mean msys2 ucrt64? That's what I use since it's the "default".
So the packages installed via pacman are sufficient. No need to install software from qt.io (as needed for a compile with Qt SDK)?
qt-cmake isn't found. What is it part of? I could only find qt-cmake.bat in some folder. But that didn't work.
Do I need to set a prefix?
I spent hours figuring out what could be done. But I didn't succeed.
I found string CMAKE_BUILD_TYPE in CMakeLists.txt. When its value is Debug then things for translations are done. I also looked through BUILD.sh for hints.

**Reply by sunderme** [reply]
you don't need qt-cmake. qt-cmake is the same as cmake, except it contains hints where qt is located. In case of default msys2, that is not needed.

If you are struggling, I would recommend to build txs with qtcreator, directly installed from qt.io.
It does not allow poppler (pdf viewer) builds, but the rest is much simpler, especially debugging.

**Comment by octaeder** [comment]
I can activate tests and txs --execute-tests runs the tests. Even so this indicates debug mode no debug output is visible in the console.

**Comment by octaeder** [comment]
The test output in txs editor shows _Config: Using QtTest library 6.4.2, Qt 6.4.2 (x86_64-little_endian-llp64 shared (dynamic) **release build**; by GCC 12.2.0), windows 10_
shouldn't it be debug release?
On the other hand being a release build explains why the tests do not write anything to the terminal

**Comment by octaeder** [comment]
But I see following output in the messages pane. Shouldn't these lines show in the terminal?

![grafik](https://user-images.githubusercontent.com/102688820/219978184-ca2ccd9d-25df-40d7-bba5-83b5d1a84f60.png)

**Comment by sunderme** [comment]
msys2 seems not to work well with qtcreator and hence with that debugger.
vcpkg.io however works reasonable well.  i hope that poppler-qt6 is accepted as package and then I can describe how to use that.

**Comment by octaeder** [comment]
to clarify, I'm not using qtcreator since there is no poppler
